frontend/playwright/e2eTests/approveSystemUserRequest.spec.ts

```diff
@@ -1,37 +1,49 @@
 import test, { expect } from '@playwright/test';
 import { ApiRequests } from '../api-requests/ApiRequests';
 import { Token } from 'playwright/api-requests/Token';
+import { LoginWithUserPage } from 'playwright/pages/loginPage';
 
-test('Avvis Systembrukerforespørsel', async ({ page }): Promise<void> => {
-  const api = new ApiRequests(); // Create an instance
-  const tokenclass = new Token();
-  const confirmUrl = await prepareSystemUserRequest(api, tokenclass);
-
-  await page.goto(confirmUrl);
-  await page.getByRole('button', { name: 'Avvis' }).click();
-  const logoutButton = page.locator('span.sr-only', { hasText: 'Logg inn/Min profil' });
-  await expect(logoutButton).toBeVisible();
-});
-
-test('Godkjenn Systembrukerforespørsel', async ({ page }): Promise<void> => {
-  const api = new ApiRequests(); // Create an instance
-  const tokenclass = new Token();
-  const confirmUrl = await prepareSystemUserRequest(api, tokenclass);
-
-  await page.goto(confirmUrl);
-  await page.getByRole('button', { name: 'Godkjenn' }).click();
-  const logoutButton = page.locator('span.sr-only', { hasText: 'Logg inn/Min profil' });
-  await expect(logoutButton).toBeVisible();
-});
-
-async function prepareSystemUserRequest(api: ApiRequests, tokenclass: Token) {
-  const payload = api.generatePayloadSystemUserRequest();
-  const scopes =
-    'altinn:authentication/systemuser.request.read altinn:authentication/systemuser.request.write';
-  const token = await tokenclass.getEnterpriseAltinnToken(scopes);
-  const endpoint = 'v1/systemuser/request/vendor';
-  const apiResponse = await api.sendPostRequest<{ confirmUrl: string }>(payload, endpoint, token);
-  return apiResponse.confirmUrl; // Return the Confirmation URL to use in the test
-}
-
-//Lag ny test: https://authn.ui.at22.altinn.cloud/authfront/ui/auth/creation - Lag ny systemtilgang
+test.describe('Godkjenn og avvis Systembrukerforespørsel', () => {
+  let token: Token;
+  let loginPage: LoginWithUserPage;
+  let api: ApiRequests;
+
+  test.beforeEach(async ({ page }) => {
+    api = new ApiRequests();
+    token = new Token();
+    loginPage = new LoginWithUserPage(page);
+  });
+
+  test('Avvis Systembrukerforespørsel', async ({ page }): Promise<void> => {
+    //Generate confirmUrl from API
+    const confirmUrl = await prepareSystemUserRequest(api, token);
+
+    await page.goto(confirmUrl);
+    await page.getByRole('button', { name: 'Avvis' }).click();
+
+    //Expect user to be logged out
+    await expect(loginPage.LOGIN_BUTTON).toBeVisible();
+    await expect(page).toHaveURL('https://info.altinn.no');
+  });
+
+  test('Godkjenn Systembrukerforespørsel', async ({ page }): Promise<void> => {
+    const confirmUrl = await prepareSystemUserRequest(api, token);
+
+    await page.goto(confirmUrl);
+    await page.getByRole('button', { name: 'Godkjenn' }).click();
+
+    //Expect user to be logged out
+    await expect(loginPage.LOGIN_BUTTON).toBeVisible();
```

Comment:
I don't think we need both a check for login button and page url? Just login page url should be enough. However, login page url is different in each environment; `https://info.at22.altinn.cloud` and `https://info.tt02.altinn.no` which we need to handle in the test

Reply:
Was thinking you might get redirected to that page and still be logged in if some other bug/weird behavior is introduced by accident
 
when you're logged out, you get redirected to info.altinn.no for both at22 and tt02, so that probably needs to be configured then? 

Tests pass for this check in tt02 and at22:
    await expect(page).toHaveURL('https://info.altinn.no');

Reply:
Aha, I realised now that redirectUrl is set to `https://altinn.no`, which will redirect to `https://info.altinn.no` 😅 I thought this was a test without redirectUrl since you end up at the login page. Can we perhaps use a receipt page in SmartCloud to test this in the future?

Reply:
Yes 👌, but need to convince @TheTechArch to run smart cloud in AT as well despite machineporten only using one test environment.